draft-ietf-jsonpath-base.md

```diff
@@ -1104,6 +1094,34 @@ Some examples:
 | `[3,4,5]` | `$[?index(@)==2]`<br>`$[?index(@)==17]` | `[5]`<br>`[]` | Select array element |
 | `{"a":{"b":{5},c:0}}` | `$[?@.b==5 && !@.c]` | `[{"b":{5},c:0}]` | Existence  |
 
+
+## Normalized Paths
+
+A Normalized Path identifies a node by providing a query that results in exactly that node. For example,
+the JSONPath expression `$.book[?(@.price<10)]` could select two values with Normalized Paths
+`$.book[3]` and `$.book[5]`.
+
+An example application of Normalized Paths is a JSONPath implementation which outputs Normalized Paths
+instead of, or in addition to, the values identified by these paths.
+
+Since bracket notation is more general than dot notation, it is used to construct Normalized Paths.
+the syntax of Normalized Paths is restricted so that there is one and only one way of representing any
+given Normalized Path.
+
+~~~~ abnf
+normalized-path           = root-selector *(S normal-index-selector)
+normal-index-selector     = "[" S (normal-quoted-member-name / normal-element-index) S "]"
+normal-quoted-member-name = %x27 *normal-single-quoted %x27 ; 'string'
+normal-single-quoted      = normal-unescaped /
+                            ESC normal-escapable
+normal-unescaped          = %x20-26 /                       ; omit control codes
+                            %x28-5B /                       ; omit '
+                            %x5D-7E                         ; omit \
+                            %xA0-FF                         ; omit control codes
+normal-escapable           = (%x75 hexchar) ;  uXXXX U+XXXX ; TODO: exclude normal-unescaped codepoints
+normal-element-index       = "0" / (DIGIT1 *DIGIT)          ; non-negative decimal integer
+~~~~
```

Comment:
... I like that ...

Reply:
So we'd have to use \u0027 for '?

Reply:
Wouldn't we also need to omit `"` also, since we want to use normalized pathes - as they are - as potential text in JSON documents ?

Reply:
> So we'd have to use \u0027 for '?

Yes.

Reply:
> Wouldn't we also need to omit `"` also, since we want to use normalized pathes - as they are - as potential text in JSON documents ?

The requirement to use Normalized Paths as text in JSON documents is new to me. So far, I have only considered a JSONPath implementation outputting Normalized Paths instead of, or in addition to, values - end of story. What use case do you have in mind?

Reply:
As I already commented here ...

https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/152/files#r796782971

charme of using `'` delimiters within JSONPath queries and resulting pathes is the possibility of having both as unescaped values in JSON documents. Use them for referencing locations in the same or other JSON documents in JSONPath context or simply document query results in testing contexts in JSON, whatever ... thus JSONPath and JSON harmonize perfectly together.

Reply:
Great. Let me build that in. Thanks!

Reply:
example ... very unlikely ...

```json
{
     "j\"org" : "name of Jörg"
}
```

JSONPath result of `$['j"org']` or something similiar might / will be  `$['j"org']`. Putting them in 
```json
{
     "JSONPath_result": "$['j"org']"
}
```
will collide with JSON's  `"`.

Reply:
Well, putting something into a JSON string will take care of that.

Reply:
The charm of using `'` is that no escaping is needed, so the path stays readable in a JSON context.

Reply:
> As I already commented here ...
> 
> https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/152/files#r796782971
> 
> charme of using `'` delimiters within JSONPath queries and resulting pathes is the possibility of having both as unescaped values in JSON documents. Use them for referencing locations in the same or other JSON documents in JSONPath context or simply document query results in testing contexts in JSON, whatever ... thus JSONPath and JSON harmonize perfectly together.

@goessner Please see the latest commit which addresses your comment.